cfn-resources/api-key/cmd/resource/resource.go

```diff
@@ -0,0 +1,543 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/secrets"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+var CreateRequiredFields = []string{constants.OrgID, constants.Description}
+var UpdateRequiredFields = []string{constants.OrgID, constants.APIUserID, constants.Description}
+var ReadRequiredFields = []string{constants.OrgID, constants.APIUserID}
+var DeleteRequiredFields = []string{constants.OrgID, constants.APIUserID}
+var ListRequiredFields = []string{constants.OrgID}
+
+const (
```

Comment:
in one of my pull request i added this as a constant, 
https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/669/files#diff-675bbdd6b375859039850b5a5c593b26ff9269e79bd955fe1f3b9933bd83b770R138